src/backend/joanie/tests/payment/test_api_credit_card.py

```diff
@@ -66,21 +68,38 @@ def test_api_credit_card_get_credit_cards_for_new_user(self):
         self.assertTrue(User.objects.filter(username=username).exists())
 
     def test_api_credit_card_get_credit_cards_list(self):
-        """Retrieve all authenticated user's credit cards is allowed."""
+        """
+        Authenticated user should be able to retrieve all his credit cards
+        with the active payment backend.
+        """
+        payment_provider = get_payment_backend()
         user = UserFactory()
         token = self.generate_token_from_user(user)
+        # Create 2 cards for the user with the active payment_provider name
         cards = CreditCardFactory.create_batch(2, owner=user)
+        # Create 1 card for the user with another payment_provider name
+        CreditCardFactory(owner=user, payment_provider="lyra")
 
         response = self.client.get(
             "/api/v1.0/credit-cards/", HTTP_AUTHORIZATION=f"Bearer {token}"
         )
+
         self.assertEqual(response.status_code, HTTPStatus.OK)
+        self.assertEqual(payment_provider.name, "dummy")
+        self.assertEqual(CreditCard.objects.filter(owner=user).count(), 3)
+        self.assertEqual(
+            CreditCard.objects.filter(
+                owner=user, payment_provider=payment_provider.name
+            ).count(),
+            2,
+        )
```

Comment:
I don't see the interest of both assertions. First they should not be made in a test testing an API endpoint, Secondly they test nothing interesting. The methd added in the manager is tested in the good module so I see no interest for this one.

Reply:
I've removed those lines. Indeed the manager method is already tested properly in its own module. My bad.